### ISSUE TITLE
Ref Copy qL: Keep connections when copying network and select the newly creatied

### ISSUE THREAD

**Issue by DevinFour** [qLab/qLib#1258]
It would be a blessing of the Ref Copy qL shelftool would also copy the connections if you Reference Copy multiple nodes, like the original Reference Copy does as well.

Would also be more useful if afterwards the new copies would be selected, instead of still the copied.

**Comment by johnnyquest** [comment]
@DevinFour Not sure what you mean by copying the connections (this function uses the original Reference Copy under the hood btw). If you can give a more concrete description, I could look into it.

As for selecting the newly created copies, I'll be putting up an update soon, it should show up in this ticket. Cheers!

**Comment by johnnyquest** [comment]
@DevinFour I'm closing this now, but if you could elaborate on what you mean by this --

> It would be a blessing of the Ref Copy qL shelftool would also copy the connections if you Reference Copy multiple nodes, like the original Reference Copy does as well

then please reopen the ticket. Just mention me in a comment so I get a notification. Cheers

**Comment by DevinFour** [comment]
@johnnyquest 
This is my first activity on github so I hope I can still comment here to notify you. @johnnyquest

Let me specify:
When you have a network. Say a copy to points with a box on the left and a grid on the right.
It doesn't copy the connections. Hope this image makes it clear:

[https://i.imgur.com/Vn9GaQf.png](url)

**Comment by johnnyquest** [comment]
Hey @DevinFour yep cool thanks, seeing it like this makes sense. I'll take a look. Thanks

**Comment by johnnyquest** [comment]
@DevinFour btw I don't recommend reference copying entire network sections like that, there's probably better approaches to build networks. I'll fix this though

**Comment by DevinFour** [comment]
Yeah I understand, but sometimes I need a copy of a network but I need so much variation and I want to delete channels kindof on the fly and dynamically, instead of creating an hda and thinking through which parameters to promote, etc.

Thank for taking a look at it. :-)

**Comment by johnnyquest** [comment]
Hmmm I'm not sure if I can fix this, the hscript command doesn't seem to copy the connections, even if I call it with all the nodes to be copied at the same time...

**Comment by DevinFour** [comment]
I see. I guess it's not possible to check how SideFX did it under the hood, is it?

**Comment by johnnyquest** [comment]
No, I don't think so.. The only scripting equivalent I'm aware of is the hscript command "opcp" with the -c (clone) flag, BUT that doesn't copy the connections if you pass it multiple nodes (like in your example). I'll leave this ticket open, though, maybe there's a way.

**Comment by johnnyquest** [comment]
Added "question/discuss" tag to mark this as something to be looked into (at some point)
